module-new/src/components/checkbox/checkbox.component.tsx

```diff
@@ -0,0 +1,130 @@
+  direction?: 'vertical' | 'horizontal';
+
+  /** A boolean flag to disable the checkbox input. (Optional) */
+  disabled?: boolean;
+
+  /** A React.ReactNode to display a label for the checkbox input. (Optional) */
+  label?: React.ReactNode;
+
+  /** A callback function (newValue: TData) => void to handle state when 'checked' is changed. (Optional) */
+  onCheckedChange?: (newValue: TData) => void;
+
+  /** A string to set a custom data-testid attribute for the checkbox container. (Optional) */
+  testId?: string;
+
+  /** A boolean flag to automatically scroll validation error messages into view. (Optional) */
+  scrollValidationErrorsIntoView?: boolean;
+}
+
+export const Checkbox = React.forwardRef(
+  <TData extends BindType>(
+    {
+      bind,
+      className,
+      checked,
+      customIndicator,
+      customIndeterminateIndicator,
+      direction,
+      disabled,
+      onCheckedChange,
+      label,
+      testId,
+      scrollValidationErrorsIntoView,
+      ...nativeProps
+    }: ICheckboxProps<TData>,
+    ref: React.ForwardedRef<HTMLButtonElement>
+  ) => {
+    const id = React.useId();
+
+    const [value, setValue /* , bindConfig */] = useBindingState(bind, { onChange: onCheckedChange, value: checked });
+
+    const onCheckedChangeInternal = React.useCallback<Required<CheckboxProps>['onCheckedChange']>(
+      newValue => {
+        setValue?.(newValue as TData);
+      },
+      [setValue]
+    );
+
+    const indicator = React.useMemo(() => {
+      switch (value) {
+        case 'indeterminate':
+          return customIndeterminateIndicator ?? <ImMinus />;
+        default:
+          return customIndicator ?? <ImCheckmark />;
+      }
+    }, [value, customIndicator, customIndeterminateIndicator]);
+
+    return (
+      <>
+        <div
+          className={concat('arm-checkbox-container', className)}
+          data-disabled={disabled}
+          data-direction={direction}
+          data-testid={testId}
+          {...nativeProps}
+        >
+          <Root
+            className="arm-checkbox"
+            disabled={disabled}
+            id={id}
+            checked={value ?? undefined}
+            onCheckedChange={onCheckedChangeInternal}
+            ref={ref}
+          >
+            <Indicator className="arm-checkbox-indicator" asChild>
+              {indicator}
+            </Indicator>
+          </Root>
+          {label && (
+            <label className="arm-checkbox-label" data-direction={direction} htmlFor={id}>
+              {label}
+            </label>
+          )}
+        </div>
+        {/* {!!bindConfig.validationErrorMessages?.length && bindConfig.shouldShowValidationErrorMessage && (
+          <ValidationErrors
+            validationErrors={bindConfig.validationErrorMessages}
+            icon={bindConfig.validationErrorIcon}
+            scrollIntoView={scrollValidationErrorsIntoView}
+          />
+        )} */}
+      </>
+    );
+  }
+);
```

Comment:
This component is missing the "gnarly cast" (see other bindable Armstrong components.) It'll need it for the bind generics to work.